core/src/main/scala/akka/persistence/dynamodb/internal/QueryDao.scala

```diff
@@ -370,17 +388,71 @@ import software.amazon.awssdk.services.dynamodb.model.QueryResponse
     client
       .query(req)
       .asScala
-      .map { response =>
+      .flatMap { response =>
         val items = response.items()
         if (items.isEmpty)
-          None
-        else
-          Some(createSerializedJournalItem(items.get(0), includePayload))
+          Future.successful(None)
+        else {
+          Future.successful(Some(createSerializedJournalItem(items.get(0), includePayload)))
+        }
       }
       .recoverWith { case c: CompletionException =>
         Future.failed(c.getCause)
       }(ExecutionContext.parasitic)
-
   }
 
+  private def followBreadcrumb(evt: SerializedJournalItem): Future[SerializedJournalItem] =
+    if (s3Fallback.nonEmpty && evt.serManifest == S3FallbackSerializer.BreadcrumbManifest) {
```

Comment:
Shouldn't it be an error if the event is a breadcrumb and the s3Fallback.isEmpty?
Could happen if they first run with s3 enabled, but then disable it later.
The breadcrumb should never be exposed to the application.